packages/next/trace/report/to-json.ts

```diff
@@ -3,11 +3,68 @@ import { batcher } from './to-zipkin'
 import { traceGlobals } from '../shared'
 import fs from 'fs'
 import path from 'path'
+import { PHASE_DEVELOPMENT_SERVER } from '../../shared/lib/constants'
 
-let writeStream: fs.WriteStream
+let writeStream: RotatingWriteStream
 let traceId: string
 let batch: ReturnType<typeof batcher> | undefined
 
+const writeStreamOptions = {
+  flags: 'a',
+  encoding: 'utf8',
+}
+class RotatingWriteStream {
+  file: string
+  writeStream!: fs.WriteStream
+  size: number
+  sizeLimit: number
+  isRotating: Promise<void> | undefined
+  constructor(file: string, sizeLimit: number) {
+    this.file = file
+    this.size = 0
+    this.sizeLimit = sizeLimit
+    this.createWriteStream()
+  }
+  private createWriteStream() {
+    this.writeStream = fs.createWriteStream(this.file, writeStreamOptions)
```

Comment:
You need to read `this.size` from file here to initialize it when starting to append to an existing file.
I would also start by writing a `\n` to cancel any not completed line here.

Reply:
The goal is to create a new file and not start appending to an existing one